main.cpp

```diff
@@ -133,256 +143,329 @@ struct CarWash
+//      {
+//          return
+//             bool readyToEat = true;
+//      }
+//     else 
+//      {      
+//         return
+//             bool readyToEat = false;
+//     }
+// }
+
+int photosynthesize(int sunLight, int water, int unkonwnFactor)
+{
+    return (sunLight > 0) ? water * unkonwnFactor : 0;
+}
+
+struct myPerson
 {
-    // height
     float height = 5.4f;
-    // age
     int age = 29;
-    // number of siblings
     int numberOfSiblings = 3;
-    // job
     std::string job = "Driver";
-    // address
     std::string address = "1 S Anywhere St";
 
-    // Nested UDT
     struct Eye
     {
-        // member variables
         std::string color = "Blue";
         bool needsGlasses = true;
         bool hasCataracts = false;
         std::string typeOfColorBlindness = "Blue/ green";
         std::string vision = "20/50";
-        // member functions
-        void open();
-        void close();
+
+        bool open(bool awake);
+        bool close(bool awake);
         void see(std::string vision, std::string typeOfColorBlindness);
     };
-    // 3 things it can do:
-    // 1) run
-    void run(int speed);
-    // 2) jump
-    void jump(int height);
-    // 3) play piano
-    void playPiano(std::string piece);
+
+    bool run(int speed);
+    int jump(int crouch, int energy);
+    bool playPiano(bool memoryRecal, int nerves);
 };
 
+bool run(int speed)
+{
+    return (speed > 5) ? true : false;
+}
+
+int jump(int crouch, int energy)
+{
+    return crouch * energy;
+}
+
+bool playPiano(bool memoryRecal, int nerves)
+{
+     return memoryRecal && nerves > 10 ? true : false; 
+}
+
+bool myPerson::Eye::open(bool awake)
+{
+    return awake ? true : false;
+}
+
+bool myPerson::Eye::close(bool awake)
+{
+    return awake == false ? true : false;
+}
+
+// void myPerson::Eye::see(std::string vision, std::string typeOfColorBlindness)
+// {
+        // I couldn't quite figure this one out.  Maybe I can get some pointers for my code review
+// }
+
 struct Gears
 {
-    // number of gears
     int numberOfGears = 5;
-    // brand 
     std::string brand = "Acme";
-    // type
     std::string type = "Sproket";
-    // condition
     std::string condition = "Good";
-    // need maintenance?
     bool needmaintenance = false;
 
-    // 3 things it can do:
-    // 1) propel the bike
-    void propelBike();
-    // 2) spin independently of wheel
-    void freeSpin();
-    // 3) shift gear ratio
-    int shiftGear(); // returns the current gear
+    int propelBike(bool pedal, int gearRatio, int pedalSpeed);
+    bool freeSpin(bool needmaintenance);
+    int shiftGear(int leftShifterValue, int rightShifterValue); 
 };
 
+int propelBike(bool pedal, int gearRatio, int pedalSpeed)
+{
+    return pedal ? gearRatio * pedalSpeed : 0;
+}
+
+bool freeSpin(bool needmaintenance)
+{
+    return needmaintenance ? false : true;
+}
+
+int shiftGear(int leftShifterValue, int rightShifterValue)
+{
+    return leftShifterValue * rightShifterValue;    
+}
```

Comment:
These are not member functions.  these are free functions.
You did not fully qualify the function name
- [ ] revise